frappe/utils/backups.py

```diff
@@ -104,14 +104,43 @@ def zip_files(self):
 
 			print('Backed up files', os.path.abspath(backup_path))
 
-	def take_dump(self):
+	def take_dump(self, encrypt=False, automated_backup=None):
 		import frappe.utils
 
 		# escape reserved characters
 		args = dict([item[0], frappe.utils.esc(item[1], '$ ')]
 			for item in self.__dict__.copy().items())
 
-		cmd_string = """mysqldump --single-transaction --quick --lock-tables=false -u %(user)s -p%(password)s %(db_name)s -h %(db_host)s > %(backup_path_db)s """ % args
+		cmd_string = ""
+		if not automated_backup:
```

Comment:
Why should `automated_backup` variable even be a part of this code? Can you not get away with just one variable i.e. `encrypt`

Reply:
We need the automated_variable for the process of backup through a cron job. the line for backups in cron job doesn't have the option encrypt that will indicate the process to encrypt the dumped SQL file from the specific site. so if we will know that the process is coming from the cron job through automated_backup variable, we will not have to check if there is an option encrypt in the command. the process will just have to check if the option for enabling the encryption for the specific site is selected and decide to encrypt the SQL file or not.